core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -230,28 +235,62 @@ public class RaftJournalSystem extends AbstractJournalSystem {
   private RaftGroup mRaftGroup;
   private RaftPeerId mPeerId;
-  private final Map<String, TransferLeaderMessage> mErrorMessages;
+  private final Map<String, TransferLeaderMessage> mErrorMessages = new ConcurrentHashMap<>();
 
   static long nextCallId() {
     return CALL_ID_COUNTER.getAndIncrement() & Long.MAX_VALUE;
   }
 
   /**
-   * @param conf raft journal configuration
+   * Creates a {@link RaftJournalSystem}.
+   * @param path where the journal will be stored
+   * @param serviceType is either MASTER_RAFT or JOB_MASTER_RAFT
    */
-  private RaftJournalSystem(RaftJournalConfiguration conf) {
-    mConf = processRaftConfiguration(conf);
-    mJournals = new ConcurrentHashMap<>();
-    mSnapshotAllowed = new AtomicBoolean(true);
-    mTransferLeaderAllowed = new AtomicBoolean(false);
-    mPrimarySelector = new RaftPrimarySelector();
-    mAsyncJournalWriter = new AtomicReference<>();
-    mErrorMessages = new ConcurrentHashMap<>();
+  public RaftJournalSystem(URI path, ServiceType serviceType) {
+    this(path,
+        NetworkAddressUtils.getConnectAddress(serviceType, ServerConfiguration.global()),
+        ConfigurationUtils.getEmbeddedJournalAddresses(ServerConfiguration.global(), serviceType));
+  }
+
+  @VisibleForTesting
+  RaftJournalSystem(URI path, InetSocketAddress localAddress,
+      List<InetSocketAddress> clusterAddresses) {
+    mPath = new File(path.getPath());
+    mLocalAddress = localAddress;
+    mClusterAddresses = clusterAddresses;
+
+    // Override election/heartbeat timeouts for single master cluster if election timeout is not
+    // set explicitly. This is to speed up single master cluster boot-up.
+    if (mClusterAddresses.size() == 1
+        && !ServerConfiguration.isSetByUser(
+        PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT)
+        && !ServerConfiguration.isSetByUser(
+        PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT)) {
+      LOG.info("Overriding election timeout to {}ms for single master cluster.",
+          SINGLE_MASTER_ELECTION_TIMEOUT_MS);
+      ServerConfiguration.set(PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT,
+          SINGLE_MASTER_ELECTION_TIMEOUT_MS);
+      ServerConfiguration.set(PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT,
+          2 * SINGLE_MASTER_ELECTION_TIMEOUT_MS);
+    }
+
+    if (!(mClusterAddresses.contains(mLocalAddress)
+        || NetworkAddressUtils.containsLocalIp(mClusterAddresses, ServerConfiguration.global()))) {
+      LOG.error("The cluster addresses ({}) must contain the local master address ({})",
+          mClusterAddresses, mLocalAddress);
+    }
+
+    long min = ServerConfiguration.getMs(PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT);
+    long max = ServerConfiguration.getMs(PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT);
+    if (min >= max) {
+      LOG.error("Min election timeout ({}ms) should be less than max election timeout ({}ms)",
```

Comment:
Maybe its better to check these when they are used, the log an error there, then change to a valid value, i.e. set max to min.

Reply:
fixed.